src/Compilers/CSharp/Portable/FlowAnalysis/AbstractRegionControlFlowPass.cs

```diff
@@ -36,7 +36,9 @@ public override BoundNode VisitLambda(BoundLambda node)
 
         public override BoundNode VisitLocalFunctionStatement(BoundLocalFunctionStatement node)
         {
-            return VisitLocalFunctionOrLambda(node.Body);
+            VisitLocalFunctionOrLambda(node.BlockBody);
+            VisitLocalFunctionOrLambda(node.ExpressionBody);
```

Comment:
>VisitLocalFunctionOrLambda(node.ExpressionBody); [](start = 12, length = 48)

I think we should not be changing this just because we have both in the node now. Unless we have a scenario for which we want the behavior to change. Do we have a scenario?

Reply:
Also, see how method bodies are handled in PreciseAbstractFlowPass.VisitMethodBodies. If we are going to visit both bodies, I would expect us to visit them in the same manner. I believe we do want to visit both for region analysis, so that WrittenInside and other sets are properly populated based on code in both bodies. We need to come up with test scenarios and find the right visitor that should be modified.  

---
In reply to: [177626096](https://github.com/dotnet/roslyn/pull/25782#discussion_r177626096) [](ancestors = 177626096)

Reply:
My recommendation is to not make any changes to control and data flow analysis along with the fix for #24650. Our primary goal for this change is to expose information through IOperation. We definitely should investigate if it makes sense to adjust control/data flow analysis for the situation of both bodies present, but we can do that later because this is a "preexisting condition". I suggest to open a follow-up issue for that.

---
In reply to: [177628448](https://github.com/dotnet/roslyn/pull/25782#discussion_r177628448) [](ancestors = 177628448,177626096)

Reply:
Filed [https://github.com/dotnet/roslyn/issues/25794](https://github.com/dotnet/roslyn/issues/25794) to follow up.

---
In reply to: [177815839](https://github.com/dotnet/roslyn/pull/25782#discussion_r177815839) [](ancestors = 177815839,177628448,177626096)